pos_sale_order: fix issue when closing session with refund order

sebastienbeau opened the pull request.

sebastienbeau committed:
pos_sale_order: fix issue when closing session with refund order

sebastienbeau commented on pos_sale_order/models/pos_session.py:
```diff
@@ -83,29 +81,43 @@ def _create_bank_statement_line(self, statement, payments):
                 key = payment.pos_sale_order_id
             else:
-                key = payment.partner_id
+                key = payment.pos_sale_order_id.invoice_ids
```
TODO
We can improve this by adding in the ref of the move line of the payment ths invoice or sale order ref

hparfr replied:
https://github.com/akretion/pos-sale-order/pull/52 ?

sebastienbeau replied:
no, what I was meaning, is that Odoo do not set anything in the field "ref" of the payment move.
So if the accountant at some point unreconciled a payment done by the POS it's super hard to reconcile it.
To improve the user experience, now I put
- the sale order name  + the invoice name (in case of transaction split by sale order)
- the invoice none split transaction, ex: cash)

sebastienbeau committed:
pos_sale_order: set ref in payment move

Native pos do not set anything into the ref of the payment move
Native account.payment set the invoice number into teh ref of the payment move
So in our POS implementation we set the invoice name and in case of splited transaction
per sale order, we set the sale order name and the invoice name

sebastienbeau commented:
ready for merge

Kev-Roche commented on pos_sale_order/models/pos_session.py:
```diff
@@ -83,29 +92,43 @@ def _create_bank_statement_line(self, statement, payments):
-        return move_lines.filtered(
+
+    def _create_bank_statement_line(self, statement, payments, record_ref):
+        vals = self._prepare_sale_statement(payments, record_ref)
```
Only one argument here
```suggestion
        vals = self._prepare_sale_statement(payments)
```

sebastienbeau commented:
/ocabot merge minor

sebastienbeau commented:
/ocabot merge patch
